### PR TITLE
Allow "alternate layouts" for the content - field plugin

### PR DESCRIPTION
**Pull request by Bakual**
Pull Request for Issue #14753 .

### Summary of Changes
This PR adds an additional, optional parameter to the field plugin tag: `{field ID, LAYOUT}`.
This allows to customize how a field gets rendered when included in the item body.

### Testing Instructions
* Copy and rename the file /components/com_field/layouts/field/render.php to /templates/protostar/html/layouts/com_fields/field/foo.php.
* Adjust the layout so you can distinguish it from the original (eg remove the label or add some text to it)
* Create a field and add some value to it.
* Insert that field into the article body using the field button (or writing the plugin tag manually) and then add a comma after the ID and add the name of your custom layout (eg "foo"). Tag should look like `{field 1, foo}`
* Check the article in frontend

### Expected result
The field should be shown with your adjusted layout

### Actual result
The standard layout is used always

### Documentation Changes Required
Don't think there is one.
I've adjusted the plugin description text to include the possible syntaxes.

### CODE REVIEW

File: administrator/language/en-GB/en-GB.plg_content_fields.ini [hunk 1]
```diff
@@ -4,4 +4,4 @@
 ; Note : All ini files need to be saved as UTF-8
 
 PLG_CONTENT_FIELDS="Content - Fields"
-PLG_CONTENT_FIELDS_XML_DESCRIPTION="This plugin allows you to display a custom field which has been inserted with the 'Button - Fields' plugin or using Syntax: {field #} directly into the editor area."
+PLG_CONTENT_FIELDS_XML_DESCRIPTION="This plugin allows you to display a custom field which has been inserted with the 'Button - Fields' plugin or using Syntax: {field #} directly into the editor area.<br><b>Possible Syntax:</b><br/><ul><li><code>{field 1}</code> will display the field with the ID 1</li><li><code>{field 1,foo}</code> will display the selected field using the alternative layout 'foo'.</li><li><code>{fieldgroup 2}</code> will display all fields within the fieldgroup with the ID 2.</li></ul>"
```
Comment (brianteeman) [inline administrator/language/en-GB/en-GB.plg_content_fields.ini]: 1. Please use strong not b
2. br tags  should not use / or @C-Lodder will get annoyed

Comment (Bakual) [inline administrator/language/en-GB/en-GB.plg_content_fields.ini]: Thanks

**Comment by Bakual** [comment]
The plugin description will look like this
description

For testing, of course also try other combinations :smile:

**Comment by laoneo** [comment]
Just an idea, would it be hard to adapt the editor button as well to support layouts?

**Comment by Bakual** [comment]
> Just an idea, would it be hard to adapt the editor button as well to support layouts?

I thought about it but haven't had the time yet / wasn't bored enough :smile: 
The hardest thing is to fetch all possible layout for a dropdown (eg like the componentlayouts field). That dropdown could also be used in the plugin parameters to set a "default" layout.

We could also just add a text field. This would be simple, but also not very useful.

The JS part itself isn't that hard, I actually was able to do something similar in my SermonSpeaker component. Given my JS skills, that should tell you how easy it is :laughing:

**Comment by franz-wohlkoenig** [comment]
I have tested this item :red_circle: unsuccessfully on 8ec549f7b343eba318786b4399dc787d09da9e13<br /><br />changed in foo.php 
- 	`<span class="field-value"><?php echo $value; ?></span>`
to 
- 	`<span class="strong"><?php echo $value; ?></span>`

get in Source Code 
`<dd class="field-entry ">`
		`<span class="field-label">Texttestfoo: </span>`
		`<span class="field-value">alternate-layout</span>`
`</dd>`

So alternate-layout isn't loaded.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15027">issues.joomla.org/tracker/joomla-cms/15027</a>.</sub>

**Comment by Bakual** [comment]
In which component did you try this (com_content -> article? or something else?) and what is th path of the foo.php you adjusted?

**Comment by franz-wohlkoenig** [comment]
com_content, path to Template given in Instructions.

**Comment by Bakual** [comment]
Strange

**Comment by franz-wohlkoenig** [comment]
will try again.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 8ec549f7b343eba318786b4399dc787d09da9e13<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15027">issues.joomla.org/tracker/joomla-cms/15027</a>.</sub>

**Comment by Quy** [comment]
Will it be documented to create the alternative layout `/templates/protostar/html/layouts/com_fields/field/foo.php` in the Help page or in the plugin?

**Comment by Bakual** [comment]
I'd expect it to be documented on docs.joomla.org if anything. Certainly not in the plugin.
The override method used here is a common Joomla feature, nothing special at all.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 8ec549f7b343eba318786b4399dc787d09da9e13<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15027">issues.joomla.org/tracker/joomla-cms/15027</a>.</sub>

**Comment by ot2sen** [comment]
I have tested this item :white_check_mark: successfully on 8ec549f7b343eba318786b4399dc787d09da9e13<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15027">issues.joomla.org/tracker/joomla-cms/15027</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after 3 successful tests.

**Pull request merged**